### ISSUE TITLE
@elcajon: I just started looking into the auto-update in this repo and the [ha-addons-edge](https://github.com/brenner-tobias/ha-addons-edge).

### ISSUE THREAD

**Issue by elcajon** [brenner-tobias/addon-cloudflared#26]
@elcajon: I just started looking into the auto-update in this repo and the [ha-addons-edge](https://github.com/brenner-tobias/ha-addons-edge).
It looks like everything is working fine in the repo, but the r[un of update in the add-on repository](https://github.com/brenner-tobias/ha-addons-edge/actions/runs/1747835669) fails. It shows `ValueError: untagged-2eea43dd07768ed394b6 is not valid SemVer string`. This untagged-2eea43dd07768ed394b6 is the same value for different runs and I have no idea what's wrong here.
Do you mind having a look?
Thanks a lot in advance.

_Originally posted by @brenner-tobias in https://github.com/brenner-tobias/addon-cloudflared/issues/25#issuecomment-1021642200_

**Comment by elcajon** [comment]
I can't answer to the original issue as it is closed. 

I guess the updater workflow looks for the in ha-addons-edge/cloudflared/config.yaml defined version within the origin repository addon-cloudflared. As the release 0.2.4 is just a pre-release it can't be found by the workflow. I am not sure how to outline a beta release within config.yaml. Try to initially insert a different version for edge, like '42a7470' (last commit). I guess it will work afterwards.

**Comment by brenner-tobias** [comment]
Thanks for your reply. I don't think this is the issue. Nevertheless, I checked and it has exactly the same result.
Do you have any other idea?

**Comment by elcajon** [comment]
Sorry I have to pass, but the error message semVer looks to me like an error in relation to the release versioning. Have you got it running in the meantime?

**Comment by brenner-tobias** [comment]
Unfortunately not. I tried a ton of things, cleaned up my releases since I thought it might be an old release that crashes the run, but nothing helped so far.
The message changed now to `ValueError:  is not valid SemVer string`, so it looks like the string of the version is now empty, no clue why.

**Comment by brenner-tobias** [comment]
I learned from this issue that I gave the action too many rights with my PAT. After restricting the specific PAT, I run into a new problem: `fatal: unable to auto-detect email address (got 'root@6a5505f65284.(none)')'` see [here](https://github.com/brenner-tobias/ha-addons-edge/runs/5064572985?check_suite_focus=true).

Since I am pretty new to the GitHub Token topic: Can you elaborate a little bit about best practices creating and using these token? Should I create a separate GitHub account with limited access and generate the token there? Also, how does it work for you regarding the email address? I have activated "Keep my email addresses private" and assume this might lead to the problems that I see in my runs.

**Comment by elcajon** [comment]
I encountered the mail problem as well, I had to change the email in GitHub to public. As far as I know there is no other option passing the mail address to the workflow. 

Regarding the PAT topic. I am using one token within my main account. This token is dedicated for repository_updater workflows. I haven't played around with other options but I assume that you would probably run in other problems when trying to use separate accounts. Nevertheless, the workflow (even when called remotely) runs locally for your repo. I can see no advantage trying to use a second account for the PAT as the token is stored securely within the repo which calls the updater workflow.
